src/decider/gatewaydecider/utils.rs

```diff
@@ -1994,9 +2015,12 @@ pub async fn get_gateway_scoring_data(
                 set_is_experiment_tag(decider_flow, experiment_tag);
             }
 
-            let payment_source = get_true_string(txn_card_info.paymentSource.clone())
+            let payment_source = get_true_string(
+                txn_card_info.paymentSource.clone().map(|s| s.peek().to_string())
```

Comment:
```suggestion
                txn_card_info.get_payment_source())
```
add a impl for `get_payment_source` in txn_card_info